frontend/src/shared/components/header/Header.tsx

```diff
@@ -57,42 +90,158 @@ export const Header = ({ routeBreadCrumb }: HeaderProps) => {
           aria-label='Open notifications'
           color='current'
           rounded='full'
-          icon={<Icon boxSize='1.5rem' as={BiBell} />}
+          icon={
+            <>
+              <Icon boxSize='1.5rem' as={BiBell} />
+              {notificationsDTO && notificationsDTO.numberOfUnread > 0 && (
+                <Box
+                  as={'span'}
+                  color={'white'}
+                  position={'absolute'}
+                  top={'2px'}
+                  right={'3px'}
+                  fontSize={'0.8rem'}
+                  bgColor={'red'}
+                  borderRadius={'25'}
+                  zIndex={9999}
+                  p={'2px'}
+                >
+                  {notificationsDTO?.numberOfUnread}
+                </Box>
+              )}
+            </>
+          }
         ></MenuButton>
         <Portal>
-          <MenuList fontSize='sm' minW={80} shadow='lg'>
-            <MenuItem>
-              <Flex direction='column' alignItems='start' maxW={80}>
-                <Text fontWeight='bold'>Patrek</Text>
-                <Text opacity={0.8} size='sm'>
-                  Lag nasad ang Diverge
-                </Text>
-              </Flex>
-            </MenuItem>
-            <MenuItem>
-              <Flex direction='column' alignItems='start' maxW={80}>
-                <Text fontWeight='bold'>MVPrinch</Text>
-                <Text opacity={0.8} size='sm'>
-                  SURPASS YOUR LIMITS
-                </Text>
-              </Flex>
-            </MenuItem>
-            <MenuItem>
-              <Flex direction='column' alignItems='start' maxW={80}>
-                <Text fontWeight='bold'>Junix</Text>
-                <Text opacity={0.8} size='sm'>
-                  useEffect hihi
-                </Text>
+          <MenuList
+            fontSize='sm'
+            minW={80}
+            shadow='lg'
+            overflowY='scroll'
+            p={15}
+            maxH={500}
+          >
+            <Tabs size='sm' variant='soft-rounded'>
+              <Flex direction={'row'}>
+                <Heading size='md' marginBottom={2}>
+                  Notifications
+                </Heading>
+
+                <Spacer />
+                <Popover placement='bottom-end'>
+                  <PopoverTrigger>
+                    <IconButton
+                      marginTop={'-0.5rem'}
+                      aria-label='options'
+                      icon={<BsThreeDots />}
+                    />
+                  </PopoverTrigger>
+                  <PopoverContent>
+                    <PopoverArrow />
+                    <PopoverHeader>Notification Settings</PopoverHeader>
+                    <PopoverBody>
+                      <Flex direction={'column'}>
+                        <Button
+                          w={'full'}
+                          onClick={() => markAllNotificationsAsRead.mutate()}
```

Comment:
Can we have a spinner for our `Notification Tabs` body like our `Tables` implementation whenever a filter mutation (e.g. Today, Last Week, etc.) is still loading?

Perhaps we can add a
```
 if (markAllNotificationsAsRead.isLoading) {
   return // Spinner wrapper & component here
 }
 ```
 
 Maybe on our updated Notifications UI after Daxu's design 💪 
 
 ![image](https://user-images.githubusercontent.com/90462635/193748753-454f9247-4ce8-4714-92a6-52e2126648ee.png)

Reply:
This is noted. Thank you for your insights and feedback mastah raol! 🚀